subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/dependency_verification.adoc

```diff
@@ -0,0 +1,849 @@
+2. you might want to add generated checksums to the list above
+3. when _updating_ dependency verification file with more secure checksums, you don't want to accidentally erase checksums
+
+[[sec:signature-verification]]
+== Verifying dependency signatures
+
+In addition to <<sec:checksum-verification,checksums>>, Gradle supports verification of signatures.
+Signatures are used to assess the _provenance_ of a dependency (it tells who signed the artifacts, which usually corresponds to who produced it).
+They are also used to assess the integrity of a dependency just like checksums.
+While enabling signature verification usually means a higher level of security, you might want to replace checksum verification with signature verification.
+
+However:
+
+- Gradle only supports verification of signatures published on remote repositories as ASCII-armored PGP files
+- Not all artifacts are published with signatures
+
+As a consequence, signature verification will often be used alongside checksum verification.
+
+=== Enabling signature verification
+
+Because verifying signatures is more expensive (both I/O and CPU wise) and harder to check manually, it's not enabled by default.
+
+Enabling it requires you to change the configuration option in the `verification-metadata.xml` file:
+
+[source,xml]
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<verification-metadata>
+   <configuration>
+      <verify-signatures>true</verify-signatures>
+   </configuration>
+</verification-metadata>
+----
+
+[[sec:understanding-signature-verification]]
+== Understanding signature verification
+
+Once signature verification is enabled, for each artifact, Gradle will:
+
+* try to download the corresponding `.asc` file
+* if it's present
+** automatically download the keys required to perform verification of the signature
+** verify the artifact using the downloaded public keys
+* if it's absent, fallback to checksum verification
+
+That is to say that Gradle verification mechanism is much stronger if signature verification is enabled than just with checksum verification.
+In particular:
+
+- if an artifact is signed with multiple keys, all of them must pass validation or the build will fail
+- if an artifact passes verification, checksums are _ignored_
+
+However, it's not because an artifact passes checksum verification that you can trust it: you need to _trust the keys_.
+
+In practice, it means you need to list the keys that you trust for each artifact, which is done by adding a `pgp` entry instead of a `sha1` for example:
+
+[source,xml]
+----
+<component group="com.github.javaparser" name="javaparser-core" version="3.6.11">
+   <artifact name="javaparser-core-3.6.11.jar">
+      <pgp value="379ce192d401ab61"/>
+   </artifact>
+</component>
+----
+
+This effectively means that you trust `com.github.javaparser:javaparser-core:3.6.11` if it's signed with the key `379ce192d401ab61`.
```

Comment:
Do you think it makes sense to hint where end-users could (should :) ) get that PGP key ids from?
Otherwise, they'll just pick the ones from the error message which might not be the best place :(

For instance, Apache Software Foundation clarifies the ways to verify releases (https://www.apache.org/info/verification.html), JUnit5 publishes keys (see https://junit.org/junit5/ , KEYS), and so on.

Reply:
added